tests/Pension/BeamtenPensionRechnerTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Finanzrechner\Tests\Pension;
+
+use Exception;
+use Finanzrechner\Pension\BeamtenPensionRechner;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class BeamtenPensionRechner
+ * @package Finanzrechner\Tests\Pension
+ */
+final class BeamtenPensionRechnerTest extends TestCase
+{
+    /** @var BeamtenPensionRechner $rechner */
+    private $rechner;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->rechner = new BeamtenPensionRechner();
+    }
+
+    public function testCalcNoPension()
+    {
+        // Bei weniger als fünf Jahren Dienstzeit wird keine Pension gezahlt
+        $this->rechner->setDienstzeitbeginn(2000);
+        $this->rechner->setPensionseintritt(2004);
+
+        $this->assertEquals(0, $this->rechner->calc(1000));
+    }
+
+    public function testCalcMinimumPension()
+    {
+        // Bei weniger als 20 Jahren Dienstzeit errechnet sich ein Pensionssatz von
+        // weniger als 35% -> Minimum der Pension ist immer 35% von den Dienstbezügen
+        $this->rechner->setDienstzeitbeginn(2000);
+        $this->rechner->setPensionseintritt(2019);
+
+        $this->assertEquals(0.35 * 1000, $this->rechner->calc(1000));
+    }
+
+    public function testCalcMaximumPension()
+    {
+        // Nach 40+ Jahren Dienstzeit errechnet sich ein Pensionssatz von größer als
+        // 71.75 % -> Maximum der Pension ist 71.75% von den Dienstbezügen
+        $this->rechner->setDienstzeitbeginn(2000);
+        $this->rechner->setPensionseintritt(2041);
+
+        $this->assertEquals(0.7175 * 1000, $this->rechner->calc(1000));
+    }
+
+    public function testCalcPension()
+    {
+        $this->rechner->setDienstzeitbeginn(2000);
+        $this->rechner->setPensionseintritt(2025);
+
+        // Pensionssatz-Faktor * Anzahl Dienstjahre / 100 * Diensbezüge
+        $pension = 1.79375 * 25 / 100 * 1000;
+
+        $this->assertEquals($pension, $this->rechner->calc(1000));
+    }
+
+    public function testThrowsIfDienstzeitOrPensionseintrittNotSet()
+    {
+        $this->expectException(Exception::class);
+        $this->rechner->calc(1000);
+    }
```

Comment:
Den test kannst du noch mal auf zwei tests aufteilen um tatsächlich die beiden Exceptions zu testen.
aktuell Testest du nur eine mögliche Bedingung für die Exception.

Außerdem könntest du noch die "ensure-> is Positive()" testen.